src/utils/sort-versions.ts

```diff
@@ -2,28 +2,56 @@ import * as semver from 'semver';
 
 import { RunnableVersion } from '../interfaces';
 
-function electronSemVerCompare(a: semver.SemVer, b: semver.SemVer) {
-  const l = a.compareMain(b);
-  if (l) return l;
-  // Electron's approach is nightly -> other prerelease tags -> stable,
-  // so force `nightly` to sort before other prerelease tags.
-  const [prea] = a.prerelease;
-  const [preb] = b.prerelease;
-  if (prea === 'nightly' && preb !== 'nightly') return -1;
-  if (prea !== 'nightly' && preb === 'nightly') return 1;
-  return a.comparePre(b);
-}
+const preTags = ['nightly', 'alpha', 'beta'];
+
+/**
+ * Sorts prerelease tags such that nightly -> alpha -> beta.
+ *
+ * @param a Prerelease tag data for the old version.
+ * @param b Prerelease tag data for the new version.
+ * @returns 0 | 1 | -1
+ */
+const preCompare = (a: string[], b: string[]) => {
+  const one = preTags.indexOf(a[0] as string);
+  const two = preTags.indexOf(b[0] as string);
+  if (one === two) {
+    // Whether the prerelease tag number is the same
+    // e.g. alpha.1 & alpha.1.
+    if (a[1] === b[1]) return 0;
+    return a[1] > b[1] ? 1 : -1;
+  }
 
-function compare(a: semver.SemVer | string, b: semver.SemVer | string) {
-  // non-semver goes first, sorted lexicographically
-  const astr = typeof a === 'string';
-  const bstr = typeof b === 'string';
-  if (astr && bstr) return (a as string).localeCompare(b as string);
-  if (astr) return -1;
-  if (bstr) return 1;
+  return one > two ? 1 : -1;
+};
 
-  // then semver, sorted newest-to-oldest
-  return -electronSemVerCompare(a as semver.SemVer, b as semver.SemVer);
+/**
+ * Custom semver comparator which takes into account Electron's prerelease
+ * tag hierarchy.
+ *
+ * Sorts in ascending order when passed to Array.sort().
+ *
+ * @param a The old Electron version.
+ * @param b The new Electron version.
+ * @returns 0 | 1 | -1
+ */
+export function semverCompare(a: string, b: string) {
+  const pA = semver.parse(a);
+  const pB = semver.parse(b);
+
+  const sameMMP = (a: semver.SemVer | null, b: semver.SemVer | null) => {
+    if (!a || !b) return false;
+
+    return a.major === b.major && a.minor === b.minor && a.patch === b.patch;
+  };
```

Comment:
```suggestion
  const sameMain = (a: semver.SemVer | null, b: semver.SemVer | null) =>
    a && b && a.compareMain(b) === 0;
```

Reply:
TIL `compareMain`!! that's exactly what i was after 🌟